### PR TITLE
feat: add preflight check on required env variables for apps/web

### LINKED ISSUE #488: [KEY-249] Use instrumentation.ts to do preflight env checks
[https://twitter.com/ErfanEbrahimnia/status/1723741584984846532](https://twitter.com/ErfanEbrahimnia/status/1723741584984846532)

<sub>From [SyncLinear.com](https://synclinear.com) | [KEY-249](https://linear.app/unkey/issue/KEY-249/use-instrumentationts-to-do-preflight-env-checks)</sub>

### PR DESCRIPTION
**Pull request by spiritanand**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation

### Before this PR

No checks for `.env` variables. Random cryptic errors thrown.

### After this PR

Checking each required variable `.env` in `app/web`. Now, if any variable is missing, throws very specific error.

eg.
image

### Related Issue (optional)

Closes #488

### CODE REVIEW

**Comment by chronark** [comment]
Let's use the zod schema from env.ts instead, that way we only have to maintain one thing and can do more validations if we want to

**Comment by spiritanand** [comment]
@chronark do we want a more formatted error message? Or will this do?

image

File: apps/web/lib/env.ts [hunk 1]
```diff
@@ -19,23 +19,26 @@ export const env = () =>
...
     })
     .parse(process.env);
 
 export const dbEnv = () =>
   z
     .object({
-      DATABASE_HOST: z.string(),
-      DATABASE_USERNAME: z.string(),
-      DATABASE_PASSWORD: z.string(),
+      DATABASE_HOST: z.string().min(1),
```
Comment (spiritanand) [inline apps/web/lib/env.ts:39]: A min length of 1 is required when parsing, because an empty `.env` variable does not throw an error.

**Comment by chronark** [comment]
Check [this](https://github.com/unkeyed/unkey/blob/178797b7ae91327130d7299a21695c863d9abaee/apps/api/src/pkg/errors/http.ts#L90). 
I found this nice package that makes the zod errors much easier to read

**Comment by spiritanand** [comment]
Looking better @chronark

**Comment by spiritanand** [comment]
> @chronark do we want a more formatted error message? Or will this do?
> 
> image

Although I do not mind such a spread out error message (very easy to decode for people familiar with zod). Your thoughts?

**Comment by chronark** [comment]
Exactly, it's easy if you are familiar with zod.
But most people running into .env issues are new to the project so we canz make the assumption they know zod well enough

Lgtm

**Pull request closed without merging**